lib/post_creator.rb

```diff
@@ -107,8 +107,8 @@ def valid?
       topic_creator = TopicCreator.new(@user, guardian, @opts)
       return false unless skip_validations? || validate_child(topic_creator)
     else
-      @topic = Topic.find_by(id: @opts[:topic_id])
-      if (@topic.blank? || !guardian.can_create?(Post, @topic))
+      @topic = Topic.with_deleted.find_by(id: @opts[:topic_id])
+      if (@topic.blank? || !guardian.can_create?(Post, @topic)) && !skip_validations?
```

Comment:
@tgxworld Both tests added here fail without this check.

For example, in the case of creating a "topic is closed" moderator post, we need to check `skip_validations?` because otherwise the previous condition will fail, since the `topic` is already closed when creating the moderator post, and thus `!guardian.can_create?(Post, @topic)` will be false (https://github.com/discourse/discourse/blob/860a7e1fc006d87ef324740f51c108d973f17e28/lib/guardian/topic_guardian.rb#L29).

In general I think all those guardian checks should be skipped whenever we create moderator posts (since regular validations don't apply for those), so that's why they pass the `skip_validations` parameter. That was the case everywhere except for here.